src/mbgl/map/map_context.cpp

```diff
@@ -113,6 +113,8 @@ void MapContext::setStyleURL(const std::string& url) {
 
         if (res.status == Response::Successful) {
             loadStyleJSON(res.data, base);
+        } else if (res.status == Response::NotFound && (styleURL.find("mapbox://") != std::string::npos)) {
```

Comment:
`styleURL.find("mapbox://") != 0`? Should match only at the beginning. Also, extra parentheses are not needed here.